pkg/controller/backup_controller_test.go

```diff
@@ -561,11 +559,20 @@ func TestProcessBackupCompletions(t *testing.T) {
 
 			// Ensure we have a CompletionTimestamp when uploading.
 			// Failures will display the bytes in buf.
-			completionTimestampIsPresent := func(buf *bytes.Buffer) bool {
-				return strings.Contains(buf.String(), `"completionTimestamp": "2006-01-02T22:04:05Z"`)
-			}
+			// completionTimestampIsPresent := func(buf *bytes.Buffer) bool {
+			// 	return strings.Contains(buf.String(), `"completionTimestamp": "2006-01-02T22:04:05Z"`)
+			// }
 			backupStore.On("BackupExists", test.backupLocation.Spec.StorageType.ObjectStorage.Bucket, test.backup.Name).Return(test.backupExists, test.existenceCheckError)
-			backupStore.On("PutBackup", test.backup.Name, mock.MatchedBy(completionTimestampIsPresent), mock.Anything, mock.Anything, mock.Anything).Return(nil)
+			// backupInfo := persistence.BackupInfo{
+			// 	Name: test.backup.Name,
+			// 	// Metadata:         mock.MatchedBy(completionTimestampIsPresent),
+			// 	// Contents: mock.Anything,
+			// 	// Log: mock.Anything,
+			// 	// PodVolumeBackups: mock.Anything,
+			// 	// VolumeSnapshots:  mock.Anything,
+			// }
+			// TODO: [carlisia] resolve this mock
+			backupStore.On("PutBackup", mock.Anything).Return(nil)
```

Comment:
I **think** what you want to do here is define a function that can be used as an arg to `mock.MatchedBy` -- so something like the following:

```go
func hasNameAndCompletionTimestamp(info persistence.BackupInfo) bool {
    return info.Name == test.backup.Name &&
               strings.Contains(info.Metadata.String(), `"completionTimestamp": "2006-01-02T22:04:05Z"`)    
}
```

And then this line becomes:

```go
backupStore.On("PutBackup", mock.MatchedBy(hasNameAndCompletionTimestamp)).Return(nil)
```